apps/server/src/services/RestoreService.ts

```diff
@@ -0,0 +1,124 @@
+            try {
+                fs.mkdirSync(this.filePath);
+            } catch (err) {
+                logger.error(LogOrigin.Server, err);
+                this.hasValidData = false;
+                return;
+            }
+        }
+
+        try {
+            const data = fs.readFileSync(path.join(this.filePath, 'restore.csv'), 'utf-8');
+            const elements = data.split(',');
+            if (elements[5] == '\n') {
+                logger.info(LogOrigin.Server, 'intact restore file');
+
+                const maybePlayback = elements[0] as Playback
+                if (!Object.values(Playback).includes(maybePlayback)) { this.hasValidData = false; return; }
+                this.playback = maybePlayback;
+
+                const maybeId = elements[1] === 'null' ? null : elements[1];
+                //TODO: Cannot access 'EventLoader' before initialization
+                // if (maybeId && EventLoader.getEventWithId(maybeId)) { this.hasValidData = false; return; }
+                this.selectedEventId = maybeId;
+
+                const maybeStartedAt = elements[2] === 'null' ? null : parseInt(elements[2]);
+                if (maybeStartedAt !== null && Number.isNaN(maybeStartedAt)) { this.hasValidData = false; return; }
+                this.startedAt = maybeStartedAt;
+
+                const maybeAddedTime = elements[3] === 'null' ? null : parseInt(elements[3]);
+                if (maybeAddedTime !== null && Number.isNaN(maybeAddedTime)) { this.hasValidData = false; return; }
+                this.addedTime = maybeAddedTime;
+
+                const maybePausedAt = elements[4] === 'null' ? null : parseInt(elements[4]);
+                if (maybePausedAt !== null && Number.isNaN(maybePausedAt)) { this.hasValidData = false; return; }
+                this.pausedAt = maybePausedAt;
+
+                this.hasValidData = true;
+            } else {
+                this.hasValidData = false;
+            }
+        } catch (err) {
+            logger.info(LogOrigin.Server, 'faild to open restore.csv, ' + err);
+            this.hasValidData = false;
+        }
+        this.file = new Writer(path.join(this.filePath, 'restore.csv'));
+    }
+
+    /**
+    * @param {Playback} playback
+    * @param {string} selectedEventId
+    * @param {number} startedAt
+    * @param {number} addedTime
+    * @param {number} pausedAt
+    */
+    async save(playback: Playback, selectedEventId: string | null, startedAt: number | null, addedTime: number | null, pausedAt: number | null) {
+        const newStore = `${playback},${selectedEventId},${startedAt},${addedTime},${pausedAt},\n`;
+        if (newStore != this.lastStore) {
+            this.lastStore = newStore;
+            this.file.write(newStore).catch((err) => { logger.error(LogOrigin.Server, err) });
+        }
+    }
+
+    /**
+    * try to restore timer state from csv
+    */
+    restore() {
+        if (!this.hasValidData) return;
+        switch (this.playback) {
```

Comment:
I am not a big fan of this use for switch cases.
Would prefer to see this as separate if statements